File: solutions/python/allergies/2/allergies.py
class Allergies:
    ingredients = (
        "eggs",
        "peanuts",
        "shellfish",
        "strawberries",
        "tomatoes",
        "chocolate",
        "pollen",
        "cats",
    )

    def __init__(self, score):
        self.score = score

    def allergic_to(self, item):
        return bool(self.score & 2 ** Allergies.ingredients.index(item))

    @property
    def lst(self):
        return [
            ingredient
            for ingredient in Allergies.ingredients
            if self.allergic_to(ingredient)
        ]

File: solutions/go/allergies/2/allergies.go
package allergies

var allergenMap = map[string]int{
	"eggs":         1,
	"peanuts":      2,
	"shellfish":    3,
	"strawberries": 4,
	"tomatoes":     5,
	"chocolate":    6,
	"pollen":       7,
	"cats":         8,
}

func Allergies(allergies uint) (result []string) {
	for allergen, _ := range allergenMap {
		if AllergicTo(allergies, allergen) {
			result = append(result, allergen)
		}
	}
	return
}

func AllergicTo(allergies uint, allergen string) bool {
	score, found := allergenMap[allergen]
	if found != true {
		return false
	}
	return allergies&(1<<(score-1)) != 0
}
